src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.classes.UserAuthenticationData;
+import com.google.sps.classes.Utility;
+import java.io.IOException;
+import java.sql.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/*
+ * Servlet that creates login or logout URL and sends it as response.
+ */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    // Set default variables to create UserAuthenticationData object.
+    Boolean loggedIn = false;
+    String email = "";
+    Boolean isUserRegistered = false;
+    String authenticationUrl = "";
+    String redirectUrl = "/";
+
+    // If user is logged in, udpate variables.
+    // Set authenticationUrl to either logout or login URL.
+    if (userService.isUserLoggedIn()) {
+      loggedIn = true;
```

Comment:
Javadoc is more readable if it is inside of the if statement
```suggestion
    if (userService.isUserLoggedIn()) {
      // If user is logged in, udpate variables.
      // Set authenticationUrl to either logout or login URL.
      loggedIn = true;
```

Reply:
But these comments describe the whole if block, so I think they should precede the if statement. Otherwise, it seems I am describing the variables below, don't you think?

Reply:
Drive by comment - putting the comment within the if statement rather than outside of it is preferred because it's easier for the reader to see that it only refers to the 'if' part of the block, and not the 'else'. In other cases, it's still good to do for consistency's sake. If you have a comment that handles both the 'if' and 'else' case, you should break it up into multiple and place it with the corresponding part of the block!

Reply:
Ooh, ok, noted! I made the change, but I don't know if the comments should be indented to the level of the if statement, or have two extra spaces, and I also don't know if I needed to include a blank line below the comment.